test/unit/alignment/pairwise/align_pairwise_test.cpp

```diff
@@ -86,6 +88,66 @@ TYPED_TEST(align_pairwise_test, single_pair)
             EXPECT_EQ(gap2 | seqan3::views::to_char | seqan3::views::to<std::string>, "A-GTGATACT");
         }
     }
+
+    // use std::pair
+
+    std::pair sequences{"ACGTGATG"_dna4,"AGTGATACT"_dna4};
+
+    {  // the score
+        seqan3::configuration cfg = seqan3::align_cfg::edit | seqan3::align_cfg::result{seqan3::with_score};
+
+        for (auto && res : call_alignment<TypeParam>(sequences, cfg))
+        {
+            EXPECT_EQ(res.score(), -4.0);
+        }
+    }
+
+    {  // the alignment
+        seqan3::configuration cfg = seqan3::align_cfg::edit | seqan3::align_cfg::result{seqan3::with_alignment};
+        unsigned idx = 0;
+        for (auto && res : call_alignment<TypeParam>(sequences, cfg))
+        {
+            EXPECT_EQ(res.id(), idx++);
+            EXPECT_EQ(res.score(), -4);
+            EXPECT_EQ(res.back_coordinate().first, 8u);
+            EXPECT_EQ(res.back_coordinate().second, 9u);
+            auto && [gap1, gap2] = res.alignment();
+            EXPECT_EQ(gap1 | seqan3::views::to_char | seqan3::views::to<std::string>, "ACGTGATG--");
+            EXPECT_EQ(gap2 | seqan3::views::to_char | seqan3::views::to<std::string>, "A-GTGATACT");
+        }
+    }
+}
+
+TYPED_TEST(align_pairwise_test, single_pair_of_non_lvalues)
+{
+    auto const seq1 = "ACGTGATG"_dna4;
+    auto const seq2 = "AGTGATACT"_dna4;
+
+    auto p = std::make_pair(seq1, seq2);
+
+    {  // the score
+        seqan3::configuration cfg = seqan3::align_cfg::edit | seqan3::align_cfg::result{seqan3::with_score};
+
+        for (auto && res : call_alignment<TypeParam>(p, cfg))
```

Comment:
But `p` is an lvalue here, no? I don't understand the test name...
In align_pairwise.hpp you coded `static_assert(std::is_lvalue_reference_v<sequence_t>)`, so why is it supposed to work with non-lvalues?

Reply:
I cherry picked this test from @smehringer, maybe she can explain why she called it that? Or should I just put it in the single_pair test above?

Reply:
sorry for the naming! I copied an pasted from the other test.

You are right that p itself is an lvalue **but** the types within the pair are not!

So with `std::tie` the type is
`std::pair<seq1_type &, seq1_type &> &`
and with make_pair it is
`std::pair<seq1_type, seq1_type> &`

Reply:
Okay, than I put it all to single_pair.

Reply:
Thanks for the explanation!